packages/quantic/force-app/main/default/lwc/quanticNotifications/quanticNotifications.js

```diff
@@ -48,32 +59,60 @@ export default class QuanticNotifications extends LightningElement {
   initialize = (engine) => {
     this.headless = getHeadlessBundle(this.engineId);
     this.notifyTrigger = this.headless.buildNotifyTrigger(engine);
+    this.searchStatus = this.headless.buildSearchStatus(engine);
     this.ariaLiveNotificationsRegion = AriaLiveRegion('notifications', this);
     this.unsubscribe = this.notifyTrigger.subscribe(() => this.updateState());
+    this.unsubscribeSearchStatus = this.searchStatus.subscribe(() =>
+      this.handleSearchStatusChange()
+    );
   };
 
   disconnectedCallback() {
     this.unsubscribe?.();
+    this.unsubscribeSearchStatus?.();
   }
 
   updateState() {
-    this.notifyTriggerState = this.notifyTrigger.state;
-    this.ariaLiveNotificationsRegion.dispatchMessage(
-      this.notifyTriggerState?.notifications.reduce(
-        (value, notification, index) => {
-          return `${value} Notification ${index + 1}: ${notification}`;
-        },
-        ''
-      )
-    );
-  }
-
-  get notifications() {
-    return (
-      this.notifyTriggerState?.notifications.map((notification, index) => ({
+    this.notifications =
+      this.notifyTrigger?.state?.notifications.map((notification, index) => ({
         value: notification,
         id: index,
-      })) ?? []
+        visible: true,
+      })) ?? [];
+
+    this.dispatchAriaLiveNotificationsRegionMessage();
+  }
+
+  handleSearchStatusChange() {
+    if (!this.searchStatus?.state.isLoading) {
+      if (this.searchStatus?.state?.hasError) {
+        this.notifications = [];
+      } else {
+        this.notifications =
+          this.notifications?.map((notification) => ({
+            ...notification,
+            visible: true,
+          })) ?? [];
+        this.dispatchAriaLiveNotificationsRegionMessage();
+      }
+    }
+  }
+
+  handleNotificationClose(event) {
+    const currentNotificationId = event.currentTarget.dataset.id;
+    this.notifications = this.notifications.map((notification) => {
```

Comment:
Very interesting point here, this method that you used will override the entire `this.notifications` state.

Which is good sometimes, bad some other times. And I can't figure out if it's good or bad here :D 

On one side, it's possible that overriding the entire `this.notifications` will "re-draw" all the notifications HTML, so it could be simplified by doing something like this instead 

`this.notifications.find(notification => notification.id === currentNotification).visible = false;`

Although I do believe that it's possible that LWC will **not** refresh the variable reactivity since you'd be changing a value deep within the state that's being watched by LWC to re-draw.

Overall I think it's good to leave as-is, but consider these two options in the future to make sure to make a conscious decision.

In short, do I change the data "in-place" (a mutation) or do I override the complete object with the new state, (immutability). In general Immutability is regarded as a good thing, especially in functional programming. Because it ensures there's no side-effect since you always change the whole object, so something cannot be "watching" the object and not get notified of changes because someone changed a deep value instead of the whole object.

Reply:
good point, I will keep this in mind!